lib/lbt/analyzer/ComponentAnalyzer.js

```diff
@@ -116,12 +117,43 @@ class ComponentAnalyzer {
 
 		// TODO usages
 
+		// See sap/ui/core/Component._createManifestModelConfigurations
 		each( ui5.models, (options, model) => {
+			let modelType;
 			if ( options.type ) {
-				const module = ModuleName.fromUI5LegacyName( options.type );
-				log.verbose("derived model implementation dependency ", module);
-				info.addDependency(module);
+				modelType = options.type;
+			} else if ( options.dataSource && sapApp.dataSources ) {
+				const oDataSource = sapApp.dataSources[options.dataSource];
+				if (!oDataSource) {
+					return;
+				}
+				// default dataSource type is OData
+				const dataSourceType = oDataSource.type || "OData";
+				switch (dataSourceType) {
+				case "OData":
+					if (oDataSource.settings && oDataSource.settings.odataVersion === "4.0") {
+						modelType = "sap.ui.model.odata.v4.ODataModel";
+					} else {
```

Comment:
not yet relevant, but should be limited to odataVersion === "2.0" or undefined.

Reply:
Good point. I'll fix it here and in the runtime.